src/main/java/com/facebook/openwifirrm/ucentral/operationelement/HTOperationElement.java

```diff
@@ -0,0 +1,206 @@
+	/**
+	 * Defines the channel center frequency for a 160 or 80+80 MHz BSS bandwidth
+	 * with NSS support less than Max VHT NSS. This is 0 for non-VHT STAs. See
+	 * 802.11 for details.
+	 */
+	private final byte channelCenterFrequencySegment2;
+	/** False if no STBC beacon is transmitted; true otherwise. */
+	private final boolean dualBeacon;
+	/** False if dual CTS protection is not required; true otherwise. */
+	private final boolean dualCtsProtection;
+	/** False in a primary beacon. True in an STBC beacon. */
+	private final boolean stbcBeacon;
+	/**
+	 * Indicates the HT-MCS values that are supported by all HT STAs in the BSS. A
+	 * bitmap where a bit is set to 1 to indicate support for that MCS and 0
+	 * otherwise, where bit 0 corresponds to MCS 0.
+	 */
+	private final byte[] basicHtMcsSet;
+
+	/**
+	 * Constructs an {@code HTOperationElement} using the given field values. See
+	 * 802.11 for more details.
+	 * <p>
+	 * For details about the parameters, see the javadocs for the corresponding
+	 * member variables.
+	 */
+	public HTOperationElement(byte primaryChannel, byte secondaryChannelOffset, boolean staChannelWidth,
+			boolean rifsMode, byte htProtection, boolean nongreenfieldHtStasPresent, boolean obssNonHtStasPresent,
+			byte channelCenterFrequencySegment2, boolean dualBeacon, boolean dualCtsProtection, boolean stbcBeacon) {
+		/*
+		 * XXX some combinations of these parameters may be invalid as defined by
+		 * 802.11-2020, but this is not checked here. If fidelity to 802.11 is required,
+		 * the caller of this method must make sure to pass in valid parameters. The
+		 * 802.11-2020 specification has more details about the parameters.
+		 */
+		this.primaryChannel = primaryChannel;
+		this.secondaryChannelOffset = secondaryChannelOffset;
+		this.staChannelWidth = staChannelWidth;
+		this.rifsMode = rifsMode;
+		this.htProtection = htProtection;
+		this.nongreenfieldHtStasPresent = nongreenfieldHtStasPresent;
+		this.obssNonHtStasPresent = obssNonHtStasPresent;
+		this.channelCenterFrequencySegment2 = channelCenterFrequencySegment2;
+		this.dualBeacon = dualBeacon;
+		this.dualCtsProtection = dualCtsProtection;
+		this.stbcBeacon = stbcBeacon;
+		// the next 16 bytes are for the basic HT-MCS set
+		// a default is chosen; if needed, we can add a parameter to set these
+		this.basicHtMcsSet = new byte[16];
+	}
+
+	/** Constructor with the most used parameters. */
+	public HTOperationElement(byte primaryChannel, byte secondaryChannelOffset, boolean staChannelWidth,
+			byte channelCenterFrequencySegment2) {
+		this(primaryChannel, secondaryChannelOffset, staChannelWidth, false, (byte) 0, true, false,
+				channelCenterFrequencySegment2, false, false, false);
+	}
+
+	/**
+	 * Constructs an {@code HTOperationElement} by decoding {@code htOper}.
+	 *
+	 * @param htOper a base64 encoded properly formatted HT operation element (see
+	 *               802.11)
+	 */
+	public HTOperationElement(String htOper) {
```

Comment:
Are you sure that B0 is supposed to be the right-most bit? It looks so from the figures in the standard but just want to be sure.

Reply:
By B0 do you mean "bit 0" of the basic MCS set field?

If so, it doesn't actually matter whether it is right-most or left-most since right now I'm just copying it bit-by-bit (I copy the bytes exactly from the decoded String into `basicHtMcsSet`) and never using it. But if it did matter, I think usually bit 0 refers to the left-most and that is also what I see in Section 9.4.2.55.4.

Reply:
Haleema and I discussed. She made a good catch. The actual operation elements come in "big endian" in terms of byte order but "little endian" in terms of bit order within a byte (alternatively, one can interpret it as all "big endian" but in that case the illustrations in the 802.11 are drawn "backwards" graphically). I will fix this in the next push.